### ISSUE TITLE
Support for React v18.x

### ISSUE THREAD

**Issue by brettdewoody** [pocka/figspec#41]
This package currently has a peer dependency of React 16|17. With Storybook [now supporting React v18](https://github.com/storybookjs/storybook/blob/82696f194790378720ecb5e946f4bb3636a9e0df/app/react/package.json#L90) can we update this to include React 18 as a peer dependency. 

I'm happy to submit a PR for this upgrade if desired.

**Comment by pocka** [comment]
Yes, we should add v18 to the allowed versions! I checked the source code of `@lit-labs/react` (which this package uses internally for Lit to React conversion) and saw no deprecated or deleted APIs. PR would be welcome.
